sources/packages/backend/apps/api/src/route-controllers/student-account-applications/_test_/e2e/student-account-application.aest.controller.approveStudentAccountApplication.e2e-spec.ts

```diff
@@ -0,0 +1,155 @@
+      firstName: user.firstName,
+      lastName: user.lastName,
+      email: user.email,
+      dateOfBirth: "2001-01-31",
+      phone: "1234567890",
+      sinNumber: "534012702",
+      mode: "aest-account-approval",
+      identityProvider: "bceidboth",
+      sinConsent: true,
+      gender: "X",
+      addressLine1: "address 1",
+      city: "Victoria",
+      country: "Canada",
+      postalCode: "H1H1H1H",
+      provinceState: "BC",
+      selectedCountry: "canada",
+      canadaPostalCode: "H1H1H1H",
+    };
+    // Save the fake student account application to be later approved by the Ministry
+    // and create the Student Account.
+    const studentAccountApplication = await db.studentAccountApplication.save(
+      createFakeStudentAccountApplication(
+        { user },
+        { initialValues: { submittedData } },
+      ),
+    );
+
+    await saveFakeSFASIndividual(db.dataSource, {
+      initialValues: {
+        lastName: user.lastName,
+        birthDate: "2001-01-31",
+        sin: "534012703",
+      },
+    });
+
+    const endpoint = `/aest/student-account-application/${studentAccountApplication.id}/approve`;
+    const token = await getAESTToken(AESTGroups.BusinessAdministrators);
+    // Mock the form.io response.
+    sharedFormService.dryRunSubmission = jest.fn().mockResolvedValue({
+      valid: true,
+      formName: FormNames.StudentProfile,
+      data: { data: submittedData },
+    });
+
+    // Act/Assert
+    await request(app.getHttpServer())
+      .post(endpoint)
+      .send(submittedData)
+      .auth(token, BEARER_AUTH_TYPE)
+      .expect(HttpStatus.CREATED)
+      .then((response) => {
+        expect(response.body.id).toBeGreaterThan(0);
+      });
+    // Check that the notification is in the database.
+    const notificationMessageType =
+      NotificationMessageType.PartialStudentMatchNotification;
+    const notification = await db.notification.findOne({
+      select: {
+        id: true,
+        dateSent: true,
+        messagePayload: true,
+        notificationMessage: { templateId: true },
+      },
+      relations: { notificationMessage: true, user: true },
+      where: {
+        notificationMessage: {
+          id: notificationMessageType,
+        },
+      },
+    });
+    expect(notification.dateSent).toBe(null);
+    expect(notification.messagePayload["email_address"]).toEqual(
+      "dummy@some.domain",
+    );
+    expect(notification.messagePayload["personalisation"]["lastName"]).toEqual(
+      user.lastName,
+    );
+    expect(
+      notification.messagePayload["personalisation"]["givenNames"],
+    ).toEqual(user.firstName);
+    expect(
+      notification.messagePayload["personalisation"]["studentEmail"],
+    ).toEqual(user.email);
+    expect(notification.messagePayload["personalisation"]["dob"]).toEqual(
+      "Jan 31 2001",
+    );
+    expect(notification.messagePayload["personalisation"]["matches"]).toEqual(
+      "Last name and birth date match.",
+    );
```

Comment:
Unless for a reason, the preference would be the comparison executed as below for better readability.
```ts
expect(notification).toEqual({
    dateSent: null,
    messagePayload: {
      email_address: "dummy@some.domain",
      personalisation: {
        lastName: user.lastName,
        givenNames: user.firstName,
        studentEmail: user.email,
        dob: "Jan 31 2001",
        matches: "Last name and birth date match.",
      },
    },
  });
```

Reply:
I didn't have an exact match for the matchdate that is part of the personalisation, or a particular way to get a matching time, so I broke out only the sections that I did have matches for.

Reply:
When there is a specific value out of our control the above can still be used as
```ts
personalisation: {
  lastName: user.lastName,
  givenNames: user.firstName,
  studentEmail: user.email,
  dob: "Jan 31 2001",
  matches: "Last name and birth date match.",
  matchdate: expect.any(String),
},
```

Reply:
updated